Suppress UnretainedCallArgs warnings in GeneratedWebKitSecureCoding.mm

fred-wang opened the pull request:
#### 08f7cb918061c363ab69710b5e38ee6d9818e119
<pre>
Suppress UnretainedCallArgs warnings in GeneratedWebKitSecureCoding.mm
<a href="https://bugs.webkit.org/show_bug.cgi?id=300502">https://bugs.webkit.org/show_bug.cgi?id=300502</a>

Reviewed by Ryosuke Niwa.

The static analysis checker warns about unretained arg for isKindOfClass, so
suppress these warnings: <a href="https://github.com/llvm/llvm-project/issues/162979">https://github.com/llvm/llvm-project/issues/162979</a>

* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations: Update expectation.
* Source/WebKit/Scripts/generate-serializers.py: Suppress warning.
(generate_webkit_secure_coding_impl): Ditto.

Canonical link: <a href="https://commits.webkit.org/301353@main">https://commits.webkit.org/301353@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1da6e76b2ec468d6eedbf999ee313c22826a34fb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125706 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45368 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36117 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132569 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77589 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46052 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53926 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95768 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/63886 "An unexpected error occured. Recent messages:") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128654 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36815 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112407 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/76260 "1 api test failed or timed out") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/125058 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35717 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/76040 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106593 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30806 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/135248 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52495 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40254 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/104234 "Found 6 new test failures: imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-002.html imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-010.html imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-inline-010.html imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-last-word-001.html imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-punctuation-001.html imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-002.html (failure)") | 
| | [❌ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52943 "") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108619 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103962 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26476 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27633 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/49736 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52390 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/58197 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51738 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55089 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53434 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebKit/Scripts/generate-serializers.py:
```diff
@@ -392,7 +392,7 @@ def type_check(self):
         if value == 'SecTrustRef':
             return f'(m_{self.type} && CFGetTypeID((CFTypeRef)m_{self.type}.get()) == SecTrustGetTypeID())'
-        return f'[m_{self.type} isKindOfClass:IPC::getClass<{value}>()]'
+        return f'[m_{self.type} isKindOfClass:retainPtr(IPC::getClass<{value}>()).get()]'
```
Oh, this is silly. We should make static analyzer ignore calls to `isKindOfClass`.

rniwa replied:
Please suppress the warning for now.

fred-wang committed:
Suppress UnretainedCallArgs warnings in GeneratedWebKitSecureCoding.mm
https://bugs.webkit.org/show_bug.cgi?id=300502

Reviewed by Ryosuke Niwa.

The static analysis checker warns about unretained arg for isKindOfClass, so
suppress these warnings: https://github.com/llvm/llvm-project/issues/162979

* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations: Update expectation.
* Source/WebKit/Scripts/generate-serializers.py: Suppress warning.
(generate_webkit_secure_coding_impl): Ditto.

Canonical link: https://commits.webkit.org/301353@main
